src/lib/private_components/layered-map-resources/alter_image.js

```diff
@@ -1,39 +1,166 @@
+         *
+         * @param {WebGL context} gl - The WebGL context to use
+         * @param {shader} shadertype - Either gl.VERTEX_SHADER or gl.FRAGMENT_SHADER
+         * @param {shaderSource} string - The shader source code
+         */
 
-    const load_texture = src =>
+        const shader = gl.createShader(shaderType)
+        gl.shaderSource(shader, shaderSource)
+        gl.compileShader(shader)
+        return shader
+    }
+
+    const createProgram = (gl, vertexShader, fragmentShader) => {
+        /**
+         * Create and link a WebGL program
+         *
+         * @param {WebGL context} gl - The WebGL context to use
+         * @param {shader} vertexShader - The compiled vertex shader to attach
+         * @param {shader} fragmentShader - The compiled fragment shader to attach
+         */
+
+        const program = gl.createProgram()
+        gl.attachShader(program, vertexShader)
+        gl.attachShader(program, fragmentShader)
+        gl.linkProgram(program)
+        return program
+    }
+
+    const bindBuffer = (gl, attribName, array) => {
+        /**
+         * Bind data to buffer and link it to a given attribute name in the
+         * current program attached to the WebGL context.
+         *
+         * @param {WebGL context} gl - The WebGL context to use
+         * @param {string} attribName - The attribute name used in the shader code
+         * @param {array} array - The array data to bind to the buffer
+         */
+
+        const program = gl.getParameter(gl.CURRENT_PROGRAM)
+
+        const newBuffer = gl.createBuffer()
+        gl.bindBuffer(gl.ARRAY_BUFFER, newBuffer)
+        gl.bufferData(gl.ARRAY_BUFFER, new Float32Array(array), gl.STATIC_DRAW)
+        const attribLocation = gl.getAttribLocation(program, attribName)
+        gl.enableVertexAttribArray(attribLocation)
+        gl.bindBuffer(gl.ARRAY_BUFFER, newBuffer)
+        gl.vertexAttribPointer(attribLocation, 2, gl.FLOAT, false, 0, 0)
+    }
+
+    const bindTexture = (gl, textureIndex, uniformName, image) => {
+        /**
+         * Bind texture to a given uniform name in the current program attached
+         * to the WebGL context.
+         *
+         * @param {WebGL context} gl - The WebGL context to use
+         * @param {int} textureIndex - The texture index to use (all computers support <= 4)
+         * @param {string} uniformName - The uniform name used in the shader code
+         * @param {image element} image - The image element to use as input data for the texture
+         */
+
+        const program = gl.getParameter(gl.CURRENT_PROGRAM)
+
+        gl.activeTexture(gl.TEXTURE0 + textureIndex)
+        gl.bindTexture(gl.TEXTURE_2D, gl.createTexture())
+        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE)
+        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE)
+        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST)
```

Comment:
Should this be a prop?

Reply:
Yes, good idea. We could have a property (maybe on the `image` layer element) which states if it should be linear og nearest data point interpolation. When thinking about it, might be that `gl.LINEAR` vs. `gl.NEAREST` will not make any difference (since it is one data point per pixel). However, the CSS property [`image-rendering`](https://developer.mozilla.org/en-US/docs/Web/CSS/image-rendering) will (where e.g. `pixelated` could be a suitable choice in some scenarios). I'll make an issue of it 📃